README.md

```diff
@@ -2,548 +2,282 @@
+- Configurable:
+  - It provides many configurations and is configurable through the `~/.rdbgrc` file
+  - See the [configuration guide](/docs/configuration.md) for more information
 
 # Installation
 
 ```
 $ gem install debug
 ```
-
-or specify `-Ipath/to/debug/lib` in `RUBYOPT` or each ruby command-line option, especially for debug this gem development.
-
 If you use Bundler, write the following line to your Gemfile.
 
-```
-gem "debug", ">= 1.0.0"
+```rb
+gem "debug", ">= 1.6.0"
 ```
 
-(The version constraint is important; `debug < 1.0.0` is an older,
-abandoned gem that is completely different from this product.)
+# Usage
 
-# HOW TO USE
+The debugger is designed to support a wide range of use cases, so you have many ways to use it.
 
-To use a debugger, roughly you will do the following steps:
+But a debugging session usually consists of 4 steps:
 
-1. Set breakpoints.
-2. Run a program with the debugger.
-3. At the breakpoint, enter the debugger console.
-4. Use debug commands.
-    * [Evaluate Ruby expressions](#evaluate) (e.g. `p lvar` to see the local variable `lvar`).
-    * [Query the program status](#information) (e.g. `info` to see information about the current frame).
-    * [Control program flow](#control-flow) (e.g. move to the another line with `step`, to the next line with `next`).
-    * [Set another breakpoint](#breakpoint) (e.g. `catch Exception` to set a breakpoint that'll be triggered when `Exception` is raised).
-    * [Activate tracing in your program](#trace) (e.g. `trace call` to trace method calls).
-    * [Change the configuration](#configuration-1) (e.g. `config set no_color true` to disable coloring).
-    * Continue the program (`c` or `continue`) and goto 3.
+1. [Activate the debugger in your program](#activate-the-debugger-in-your-program)
+1. [Set breakpoints](#set-breakpoints)
+    - Through [`binding.break`](#the-bindingbreak-method)
+    - Or [breakpoint commands](#breakpoint)
+1. Execute/continue your program and wait for it to hit the breakpoints
+1. [Start debugging](#start-debugging)
+    - Here's the [full command list](#console-commands)
+    - You can also type `help` or `help <command>` in the console to see commands
 
-## Invoke with the debugger
 
-There are several options for (1) and (2). Please choose your favorite way.
+### Remote debugging
 
-### Modify source code with [`binding.break`](#bindingbreak-method) (similar to `binding.pry` or `binding.irb`)
+You can connect the debugger to your program remotely through UNIX socket or TCP/IP.
+To learn more, please check the [remote debugging guide](/docs/remote_debugging.md).
 
-If you can modify the source code, you can use the debugger by adding `require 'debug'` at the top of your program and putting [`binding.break`](#bindingbreak-method) method into lines where you want to stop as breakpoints like `binding.pry` and `binding.irb`.
+### VSCode and Chrome Devtools integration
```

Comment:
I think if we can add images or even gifs for the integrations here, it'll get people's attention to trying them out.